src/framework/theme/components/autocomplete/autocomplete.directive.ts

```diff
@@ -0,0 +1,417 @@
+        startWith(this.autocomplete.options),
+        switchMap((options: QueryList<NbOptionComponent<T>>) => {
+          return merge(...options.map(option => option.click));
+        }),
+        takeUntil(this.destroy$),
+      )
+      .subscribe((clickedOption: NbOptionComponent<T>) => this.handleOptionClick(clickedOption));
+  }
+
+  protected subscribeOnPositionChange() {
+    this.positionStrategy.positionChange
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((position: NbPosition) => {
+        this.autocomplete.overlayPosition = position;
+        this.cd.detectChanges();
+      });
+  }
+
+  protected getActiveItem(): NbOptionComponent<T> {
+    return this.keyManager.activeItem;
+  }
+
+  protected setupAutocomplete() {
+    this.autocomplete.setHost(this.hostRef);
+  }
+
+  protected getDisplayValue(value: string) {
+    const displayFn = this.autocomplete.handleDisplayFn;
+    return displayFn ? displayFn(value) : value;
+  }
+
+  protected getContainer() {
+    return this.overlayRef && this.isOpen && <ComponentRef<any>> {
+      location: {
+        nativeElement: this.overlayRef.overlayElement,
+      },
+    };
+  }
+
+  protected handleOptionClick(option: NbOptionComponent<T>) {
+
+    this.setHostInputValue(option.value);
+    this._onChange(option.value);
+
+    this.hostRef.nativeElement.focus();
+    this.autocomplete.emitSelected(option.value);
+    this.hide();
+  }
+
+  protected subscribeOnTriggers() {
+
+    this.triggerStrategy.show$
+      .pipe(filter(() => this.isClosed))
+      .subscribe(($event: Event) => {
+        this.show();
+      });
+
+    this.triggerStrategy.hide$
+      .pipe(filter(() => this.isOpen))
+      .subscribe(($event: Event) => {
+        this.hide();
+      });
+  }
+
+  protected createTriggerStrategy(): NbTriggerStrategy {
+    return this.triggerStrategyBuilder
+      .trigger(NbTrigger.FOCUS)
+      .host(this.hostRef.nativeElement)
+      .container(() => this.getContainer())
+      .build();
+  }
+
+  protected createKeyManager(): void {
+    this.keyManager = this.activeDescendantKeyManagerFactoryService
+                        .create(this.autocomplete.options);
+  }
+
+  protected setHostInputValue(value) {
+    this.hostRef.nativeElement.value = this.getDisplayValue(value);
+  }
+
+  protected createPositionStrategy(): NbAdjustableConnectedPositionStrategy {
+    return this.positionBuilder
+      .connectedTo(this.hostRef)
+      .position(NbPosition.BOTTOM)
+      .offset(0)
+      .adjustment(NbAdjustment.VERTICAL);
+  }
+
+  protected subscribeOnOverlayKeys(): void {
+    this.overlayRef.keydownEvents()
+      .pipe(
+        takeUntil(this.destroy$),
+      )
+      .subscribe((event: KeyboardEvent) => {
+        if (event.keyCode === ESCAPE && this.isOpen) {
+          this.hostRef.nativeElement.focus();
+          this.hide();
+
+        } else if (event.keyCode === ENTER) {
+          const activeItem = this.getActiveItem();
+          if (!activeItem) {
```

Comment:
Is it possible?

Reply:
Yes, if we press ESC without navigating to option in list.